### PR TITLE
HG-757 - made lazy load image processing async

### PR DESCRIPTION
**Pull request by katanka**

### CODE REVIEW

File: front/scripts/main/views/ArticleView.ts [hunk 1]
```diff
@@ -151,7 +151,15 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.LanguagesMixin, App.ViewportM
 			this.loadTableOfContentsData();
 			this.handleInfoboxes();
 			this.handlePortableInfoboxes();
-			this.lazyLoadMedia(model.get('media'));
+
+			if ( Mercury.Utils.VariantTesting.getExperimentVariationNumber({prod: '0', dev: '3066501061'}) == 1 ) {
+				Ember.run.later(this, function() {
```
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: You can use a `()=>{}` function now

File: front/scripts/main/views/ArticleView.ts [hunk 2]
```diff
@@ -151,7 +151,15 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.LanguagesMixin, App.ViewportM
 			this.loadTableOfContentsData();
 			this.handleInfoboxes();
 			this.handlePortableInfoboxes();
-			this.lazyLoadMedia(model.get('media'));
+
+			if ( Mercury.Utils.VariantTesting.getExperimentVariationNumber({prod: '0', dev: '3066501061'}) == 1 ) {
```
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: use strict equality `===`

File: front/scripts/main/views/ArticleView.ts [hunk 3]
```diff
@@ -151,7 +151,15 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.LanguagesMixin, App.ViewportM
...
+				Ember.run.later(this, function() {
+			 		this.replaceMediaPlaceholdersWithMediaComponents(model.get('media'), -1); // Process the images async
+				}, 0);
+			} else {
+				this.replaceMediaPlaceholdersWithMediaComponents(model.get('media'), -1); // Process the images synchronously
```
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: no EOL comments

File: front/scripts/main/views/ArticleView.ts [hunk 4]
```diff
@@ -184,12 +192,17 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.LanguagesMixin, App.ViewportM
...
 
-	lazyLoadMedia: function (model: typeof App.ArticleModel): void {
-		var lazyImages = this.$('.article-media');
+	replaceMediaPlaceholdersWithMediaComponents: function (model: typeof App.ArticleModel, endIndex: number): void {
+		var $mediaPlaceholders = this.$('.article-media');
 
-		lazyImages.each((index: number, element: HTMLImageElement): void => {
-			this.$(element).replaceWith(this.createMediaComponent(element, model));
-		});
+		if (endIndex == -1 || endIndex > $mediaPlaceholders.length) {
```
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: `===`

File: front/scripts/main/views/ArticleView.ts [hunk 5]
```diff
@@ -184,12 +192,17 @@ App.ArticleView = Em.View.extend(App.AdsMixin, App.LanguagesMixin, App.ViewportM
...
+			endIndex = $mediaPlaceholders.length;
+		}
+
+		// This will not iterate over components that were already replaced, since they will no longer have the 'article-media' class
+		for (var i = 0; i < endIndex; i++) {
```
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: variable declarations at the top of file

File: front/scripts/main/views/ArticleView.ts (hunk 2 above)
Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: No space after opening parens

Comment (kenkouot) [inline front/scripts/main/views/ArticleView.ts]: You can also just reference `M.VariantTesting`

**Comment by rogatty** [comment]
@katanka: this PR should be targeted at dev branch, release-90 doesn't exist any longer.

File: front/scripts/main/views/ArticleView.ts (hunk 3 above)
Comment (rogatty) [inline front/scripts/main/views/ArticleView.ts]: Should both calls have `-1` as `endIndex`?

Comment (katanka) [inline front/scripts/main/views/ArticleView.ts]: Yes, when an element is processed it loses the `article-media` class and won't be selected the next time `this.$('.article-media')` is called.

**Pull request closed without merging**